coordinator/src/main/java/org/apache/uniffle/coordinator/ApplicationManager.java

```diff
@@ -145,12 +145,12 @@ public void registerApplicationInfo(String appId, String user) {
     }
   }
 
-  public void refreshAppId(String appId) {
-    String user = appIdToUser.get(appId);
+  public void refreshAppId(String appId, String user) {
     // compatible with lower version clients
-    if (user == null) {
```

Comment:
> How to handle the case `user == null`? Because there may be existing legacy clients.

When the user field is not specified in legacy clients, the value will be empty instead of null. So if coordinator restarts, it actually calls registerApplicationInfo(appId, "").